### PR TITLE
Daily weather screen

### PR DESCRIPTION
**Pull request by atorrico09**
Forecast for next days screen (using mock images)

image

### CODE REVIEW

File: app/src/main/java/com/distillery/interview/ui/daily_weather/DailyWeatherAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import com.distillery.interview.R
+import com.distillery.interview.data.models.Daily
+import com.distillery.interview.databinding.ItemDailyWeatherBinding
+import com.distillery.interview.util.toDate
+
+class DailyWeatherAdapter : RecyclerView.Adapter<DailyViewHolder>() {
+
+    private val items = ArrayList<Daily>()
+
+    fun setItems(items: ArrayList<Daily>) {
```
Comment (huanachin) [inline app/src/main/java/com/distillery/interview/ui/daily_weather/DailyWeatherAdapter.kt:16]: Could we use DiffUtill?

File: app/src/main/java/com/distillery/interview/ui/daily_weather/DailyWeatherFragment.kt [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        setupRecyclerView()
+    }
+
+    private fun setValues(dailyWeatherResponse: DailyWeatherResponse) {
+        dailyWeatherAdapter.setItems(dailyWeatherResponse.daily as ArrayList<Daily>)
+    }
+
+    private fun setupRecyclerView() {
+        dailyWeatherAdapter = DailyWeatherAdapter()
+        binding.rvDailyWeather.layoutManager = LinearLayoutManager(
```
Comment (huanachin) [inline app/src/main/java/com/distillery/interview/ui/daily_weather/DailyWeatherFragment.kt:59]: Can we put this attribute in the XML?

File: app/src/main/java/com/distillery/interview/ui/daily_weather/DailyWeatherFragment.kt [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.distillery.interview.data.CoroutinesDispatcherProvider
+import com.distillery.interview.data.DependencyProvider
+import com.distillery.interview.data.WeatherRepository
+import com.distillery.interview.data.models.Daily
+import com.distillery.interview.data.models.DailyWeatherResponse
+import com.distillery.interview.data.models.Result
+import com.distillery.interview.databinding.FragmentDailyWeatherBinding
+
+class DailyWeatherFragment : Fragment() {
```
Comment (huanachin) [inline app/src/main/java/com/distillery/interview/ui/daily_weather/DailyWeatherFragment.kt]: Could we create a base fragment, and add some methods in common?

File: app/src/main/java/com/distillery/interview/data/source/remote/WeatherRemoteDataSource.kt [hunk 4]
```diff
@@ -28,4 +29,13 @@ class WeatherRemoteDataSource : WeatherDataSource {
                 Result.Error(e)
             }
         }
+
+    override suspend fun getDailyWeather(): Result<DailyWeatherResponse> =
+        withContext(Dispatchers.IO) {
```
Comment (lucaslabs) [inline app/src/main/java/com/distillery/interview/data/source/remote/WeatherRemoteDataSource.kt:34]: Noice!

File: app/src/test/java/com/distillery/interview/ui/DailyWeatherViewModelTests.kt [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+        whenever(weatherRepository.getDailyWeather()).thenReturn(
+            Result.Success(mockDailyWeatherResponse)
+        )
+
+        //WHEN: I open my app or move to weather screen
+        viewModel.getDailyWeather()
+
+        //THEN: I should to get weather successfully
+        Assert.assertEquals(
```
Comment (lucaslabs) [inline app/src/test/java/com/distillery/interview/ui/DailyWeatherViewModelTests.kt]: What about loading state?

File: app/src/test/java/com/distillery/interview/ui/DailyWeatherViewModelTests.kt [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+        whenever(weatherRepository.getDailyWeather()).thenReturn(
+            Result.Error(mockDailyWeatherResponse)
+        )
+
+        //WHEN: I open my app or move to weather screen
+        viewModel.getDailyWeather()
+
+        //THEN: I should to get weather unsuccessfully
+        Assert.assertEquals(
```
Comment (lucaslabs) [inline app/src/test/java/com/distillery/interview/ui/DailyWeatherViewModelTests.kt]: What about loading state?

**Review by lucaslabs — CHANGES_REQUESTED** [review]

**Review by huanachin — APPROVED** [review]

**Review by lucaslabs — APPROVED** [review]

**Pull request merged**